components/backends/trtllm/src/dynamo/trtllm/utils/api_drafter.py

```diff
@@ -0,0 +1,120 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import logging
+import os
+from typing import Dict, List
+
+from tensorrt_llm._torch.speculative.external_api import APIDrafter
+
+from dynamo.runtime import DistributedRuntime
+from dynamo.runtime.logging import configure_dynamo_logging
+
+configure_dynamo_logging()
+# TODO: remove this
+logging.getLogger().setLevel(logging.WARNING)
+
+
+class DynamoAPIDrafter(APIDrafter):
+    """
+    Custom Dynamo drafter to support internal Dynamo endpoints instead of only HTTP endpoints.
+    """
+
+    def __init__(self, spec_config, runtime: DistributedRuntime):
+        super().__init__(spec_config)
+        self.client = None
+        self.max_draft_len = spec_config.max_draft_len
+        # TODO: allow custom etcd connection info to be set in the spec_config
+        self.connection_info: Dict[str, str] = {}
+
+    async def _create_client(self):
+        try:
+            # parse endpoint
+            endpoint_path = self.endpoint.replace("dyn://", "")
+            parts = endpoint_path.split(".")
+            if len(parts) != 3:
+                raise ValueError(
+                    f"Invalid Dynamo endpoint format. Received: {self.endpoint}, but expected: dyn://namespace.component.endpoint"
+                )
+            namespace, component, endpoint = parts
+
+            # create minimal runtime for client access only
```

Comment:
Why is this necessary? why not create an endpoint client like this?
https://github.com/ai-dynamo/dynamo/blob/e22f84ab183e3b02ac34745dbcee3b9940187c35/components/backends/trtllm/src/dynamo/trtllm/main.py#L84